replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/persistence/ReplicationPersistentStore.java

```diff
@@ -82,12 +82,20 @@ public ReplicationPersistentStore(PersistentStore persistentStore) {
             String.format("Found two objects in persistence of type %s with ID %s", typeName, id));
       }
 
-      T persistable = createPersistable(klass);
-      persistable.fromMap(PersistentItem.stripSuffixes(storedMaps.get(0)));
-      return persistable;
+      T result = getPersistableFromMap(klass, storedMaps.get(0));
+      addModifiedTimeIfMissing(result);
+      return result;
     }
   }
 
+  private <T extends AbstractPersistable> T addModifiedTimeIfMissing(T result) {
+    if (result.getModified() == null) {
+      result.setModified();
+      save(result);
```

Comment:
✏️ I would not r-save it here. Leave it up to when it needs to really be modified.